### DISCUSSION TITLE
Power Assembly Param not recognized

### DISCUSSION THREAD

**Discussion by iiyoke** [SimDynamX/SpaceTeamsPRO#69] [2024 Lunar SAR Competition Q&A]
When trying to load Task 2, I keep getting the error message

power_assembly: st.Entity = st.GetSimEntity().GetParam(st.VarType.entityRef, "PowerAssembly")
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Param Exception in Entity "SimEntity", Key "PowerAssembly": Get Nonexistent Param: GetParam with type: EntityRef.

I am assuming this entity is defined in the spaceteams, but my computer can't find it. What can I do?

In fact multiple params are undefined, such as Cameras for both LTVs. However, I have updated to the newest version of spaceteams. What am I missing?

I also ensured I am on the latest version of SpaceTeams (27.5)

**Comment by connorjak** [comment]
I think your repo is not up to date with the v1.2.0 STU_Lunar_SAR code (https://github.com/SimDynamX/STU_Lunar_SAR) (should 100% match with the exception of MissionManager and EntityBehavior python files, and changes to the Task1/Task2 sim configs to adjust the Source paths to match your repo's names).

**Comment by connorjak** [comment]
In particular I recommend taking the latest example code and reapplying your changes or added code to it instead of attempting to copy only the changed portions from the new example code onto your own code.

**Comment by connorjak** [comment]
If you think your repo is up to date, please post your sim config.

**Comment by connorjak** [comment]
We just ran into the same error when trying to load Task 1 after updating things for Task 2:
```

2024-11-15T07:14:13 [FATAL]----[PySys_CompetitionBackend]: Traceback (most recent call last):
  File "C:\Users\conno\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Competition2\CompetitionBackend.py", line 34, in <module>
    power_assembly: st.Entity = st.GetSimEntity().GetParam(st.VarType.entityRef, "PowerAssembly")
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Param Exception in Entity "SimEntity", Key "PowerAssembly": Get Nonexistent Param: GetParam with type: EntityRef.
```

This should work fine when running Task 2 instead of Task 1.
